File: molecular/src/main/java/com/hartwig/actin/molecular/orange/evidence/matching/HotspotMatching.java
package com.hartwig.actin.molecular.orange.evidence.matching;

import com.hartwig.hmftools.datamodel.purple.PurpleVariant;
import com.hartwig.serve.datamodel.hotspot.VariantHotspot;

import org.jetbrains.annotations.NotNull;

public final class HotspotMatching {

    private HotspotMatching() {
    }

    public static boolean isMatch(@NotNull VariantHotspot hotspot, @NotNull PurpleVariant variant) {
        boolean geneMatch = hotspot.gene().equals(variant.gene());
        boolean chromosomeMatch = hotspot.chromosome().equals(variant.chromosome());
        boolean positionMatch = hotspot.position() == variant.position();
        boolean refMatch = hotspot.ref().equals(variant.ref());
        boolean altMatch = hotspot.alt().equals(variant.alt());

        return geneMatch && chromosomeMatch && positionMatch && refMatch && altMatch;
    }
}

File: molecular/src/main/java/com/hartwig/actin/molecular/orange/evidence/matching/HotspotMatching.kt
package com.hartwig.actin.molecular.orange.evidence.matching

import com.hartwig.hmftools.datamodel.purple.PurpleVariant
import com.hartwig.serve.datamodel.hotspot.VariantHotspot

object HotspotMatching {

    fun isMatch(hotspot: VariantHotspot, variant: PurpleVariant): Boolean {
        val geneMatch = hotspot.gene() == variant.gene()
        val chromosomeMatch = hotspot.chromosome() == variant.chromosome()
        val positionMatch = hotspot.position() == variant.position()
        val refMatch = hotspot.ref() == variant.ref()
        val altMatch = hotspot.alt() == variant.alt()

        return geneMatch && chromosomeMatch && positionMatch && refMatch && altMatch
    }
}
